src/utils/utils-vega.js

```diff
@@ -173,7 +175,11 @@ export function createVegaAttrMixin(
       rtnVal = layerObj[nullFunc]()
     } else if (input !== undefined && useScale) {
       const encodingAttrName = capAttrMap[capAttrName]
-      const capAttrObj = layerObj.getState().encoding[encodingAttrName]
+      let capAttrObj = layerObj.getState().encoding[encodingAttrName]
+      const scaleType = capAttrObj.type
```

Comment:
seems like we should protect against the case that `capAttrObj` is `undefined` for the attr unless at this point we are assured that the `encoding` lookup by default returns `{}` for an attr.